### PR TITLE
refactor: move spec processing method to vutils-extension

### PR DESCRIPTION
**Pull request by zamhown**
[[中文版模板 / Chinese template](https://github.com/VisActor/VChart/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [x] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
#2514

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vutils-extension/src/spec/transform.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import type { ISeriesSpec } from '../../typings';
+import type { ISeriesSpec } from '@visactor/vchart-types';
```
Comment (xile611) [inline packages/vutils-extension/src/spec/transform.ts:1]: `vutils-extension` 对 `@visactor/vchart-types ` 没有依赖关系
不能这么用

把基础的 `ISeriesSpec` 挪到 `vutils-extension`中，vchart可以`extends` 这个基础类型

File: packages/vutils-extension/src/spec/background.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { pickWithout } from '@visactor/vutils';
-import type { IBackgroundSpec, IBackgroundStyleSpec } from '../../typings';
+import type { IBackgroundSpec, IBackgroundStyleSpec } from '@visactor/vchart-types';
```
Comment (xile611) [inline packages/vutils-extension/src/spec/background.ts:2]: vutils-extension 对 @visactor/vchart-types  没有依赖关系
不能这么用

`@visactor/vchart-types` 是给外部导出的类型包，理论上是根据 vchart  和 vutils-extension 生成的
这个依赖顺序不对

应该把 `IBackgroundSpec`, `IBackgroundStyleSpec` 挪到`vutils-extension ` 中

File: packages/vutils-extension/package.json [hunk 3]
```diff
@@ -50,7 +51,8 @@
     "ts-loader": "8.0.2",
     "ts-node": "10.9.0",
     "tslib": "2.3.1",
-    "tslint": "5.12.1"
+    "tslint": "5.12.1",
+    "@visactor/vchart-types": "workspace:1.10.3"
```
Comment (xile611) [inline packages/vutils-extension/package.json:55]: 不能这么依赖

File: packages/vutils-extension/src/spec/transform.ts (hunk 1 above)
Comment (zamhown) [reply packages/vutils-extension/src/spec/transform.ts:1]: 引用到vchart类型的相关方法，看了下可能还是比较适合先迁回去？不太影响这次的需求。
IBackgroundSpec 和 ISeriesSpec 牵涉的基础类型比较多，包括了 mark spec 和交互等比较底层的类型，可以用泛型传入但是会让类型更复杂化

**Review by xile611 — APPROVED** [review]

**Pull request merged**